practicas/Aleksei115/P1/1_niveles_linguisticos.py

```diff
@@ -0,0 +1,554 @@
+    dict
+        Lang codes as keys and dictionary with words-transcriptions
+        as values
+    """
+    return {code: parse_response(download_ipa_corpus(code)) for code in iso_lang_codes}
+
+
+def get_ipa_transcriptions(word: str, dataset: dict) -> list[str]:
+    """Search for a word in an IPA phonetics dict
+
+    Given a word this function return the IPA transcriptions
+
+    Parameters:
+    -----------
+    word: str
+        A word to search in the dataset
+    dataset: dict
+        A dataset for a given language code
+
+    Returns
+    -------
+    list[str]:
+        List with posible transcriptions if any,
+        else an empty list
+    """
+    return dataset.get(word.lower(), "").split(", ")
+
+data = get_corpora()
+
+
+# In[19]:
+
+
+data = get_corpora()
+
+
+# In[41]:
+
+
+N_SIMILAR = 3
+def aproximacion_fonetica(funcion_simitud):
+  '''
+  Función que regresa aproximaciones de transcripciones fonéticas para palabras
+  que no están en el lexicón
+
+  Input:
+    funcion_simitud: funcion a usar para encontrar las palabras más
+    similares.
+
+  Output:
+    Regresa la transcripción fonética, ya sea la que le pertence o una
+    aproximación.
+  '''
+  rprint(Panel(Text("Representación fonética de palabras", style="bold", justify="center")))
+  rendable_langs = [Panel(get_formated_string(code, lang), expand=True) for code, lang in lang_codes.items()]
+  rprint(Columns(rendable_langs))
+
+  lang = input("lang>> ")
+  # lang = 'es_MX'
+  rprint(f"Selecciona un idioma: {lang_codes[lang]} \n\tSi quieres salir solo pon X.") if lang else rprint("Adios 👋🏼")
+  while lang:
+      sub_dataset = data[lang]
+      query = input(f"  [{lang}]word>> ")
+
+      if query=='X':
+        break;
+
+      results = get_ipa_transcriptions(query, sub_dataset)
+
+      if results == ['']: # En caso que no exista su transcripción fonética
+        print(f"Las {N_SIMILAR} aproximaciones fonéticas:")
+
+        palabras_similares = funcion_simitud(query, sub_dataset)
+        # print(palabras_similares)
+
+        for palabra_similar in palabras_similares:
+          results = get_ipa_transcriptions(palabra_similar, sub_dataset)
+          rprint(palabra_similar, " | ", ", ".join(results))
+
+      else:
+        rprint(query, " | ", ", ".join(results))
+
+  rprint("Adios 👋🏼")
+  return
+
+
+def jaccard_similarity(word1, word2, n=2):
+    set1 = set(ngrams(word1,n))
+    set2 = set(ngrams(word2,n))
+    return 1 - jaccard_distance(set1,set2)
+
+def get_similar_words_jaccard(query, dataset, n_similar=3):
+    """
+    Regresa las n palabras más similares con la distancia de Jaccard.
+
+    Input:
+      query: palabra que no está en el dataset.
+      dataset: conjunto de datos del idioma ingresado (diccionario palabra: fonemas).
+      similar: número de palabras similares a devolver (por defecto 3).
+
+    Output:
+      Regresa las n palabras más similares con la query.
+    """
+    similarities = []
+    for word, code in dataset.items():
```

Comment:
Acá no veo que uses la variable `code` dentro del ciclo. para traer solo las palabras podrías utilizar `dataset.keys()` o si no utilizas alguna variable puedes usar esta convención:


```suggestion
    for word, _ in dataset.items():
```